griptape/drivers/prompt/google_prompt_driver.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import annotations
+from collections.abc import Iterator
+from typing import TYPE_CHECKING, Optional, Any
+from attr import define, field, Factory
+from griptape.utils import PromptStack, import_optional_dependency
+from griptape.artifacts import TextArtifact
+from griptape.drivers import BasePromptDriver
+from griptape.tokenizers import GoogleTokenizer, BaseTokenizer
+
+if TYPE_CHECKING:
+    from google.generativeai import GenerativeModel
+    from google.generativeai.types import ContentDict
+
+
+@define
+class GooglePromptDriver(BasePromptDriver):
+    """
+    Attributes:
+        api_key: Google API key.
+        model: Google model name.
+        model_client: Custom `GenerativeModel` client.
+        tokenizer: Custom `GoogleTokenizer`.
+        top_p: Optional value for top_p.
+        top_k: Optional value for top_k.
+    """
+
+    api_key: Optional[str] = field(default=None, kw_only=True, metadata={"serializable": True})
+    model: str = field(kw_only=True, metadata={"serializable": True})
+    model_client: Any = field(default=Factory(lambda self: self._default_model_client(), takes_self=True), kw_only=True)
```

Comment:
Need this as `Any` instead of `GenerativeModel` or else `GoogleStructureConfig` serialization will fail.